google_gke/cluster.tf

```diff
@@ -289,7 +284,15 @@ resource "google_container_node_pool" "pools" {
     workload_metadata_config {
       mode = "GKE_METADATA"
     }
-    taint = local.node_pools_taints[each.key]
+    dynamic "taint" {
+      for_each = local.node_pools_taints
+
+      content {
+        key    = each.value.key
+        value  = each.value.value
+        effect = each.value.effect
+      }
+    }
```

Comment:
I think as I mentioned I had to use a lifecycle setting here when testing locally but let's give it a shot and see what happens.